std/algorithm/searching.d

```diff
@@ -4192,9 +4192,26 @@ bool startsWith(alias pred = "a == b", R, E)(R doesThisStart, E withThis)
 if (isInputRange!R &&
     is(typeof(binaryFun!pred(doesThisStart.front, withThis)) : bool))
 {
-    return doesThisStart.empty
-        ? false
-        : binaryFun!pred(doesThisStart.front, withThis);
+    if (doesThisStart.empty)
+        return false;
+
+    // auto-decoding special case
+    static if (isNarrowString!R)
+    {
+        dchar front;
+
+        // specialize for ASCII as to not change previous behavior
+        if (withThis <= 0x7F)
```

Comment:
Idea, if `is(E == char)`, you can skip this `if` altogether, which should be even faster for the most common case.